pageserver/src/http/routes.rs

```diff
@@ -205,7 +206,10 @@ async fn timeline_list_handler(request: Request<Body>) -> Result<Response<Body>,
         Ok::<_, anyhow::Error>(tenant_mgr::get_tenant(tenant_id, true)?.list_timelines())
     })
     .await
-    .map_err(ApiError::from_err)??;
+    // `await`-ing the task failed
+    .map_err(ApiError::from_internal_err)?
+    // `get_tenant` failed, which only happens when there's no tenant with that id
+    .map_err(ApiError::NotFound)?;
```

Comment:
This is awkwardly far away from where the `get_tenant` call. Can it be refactored into something more like:
```
let tenant = tenant_mgr::get_tenant(tenant_id, true)
    .map_err(ApiError::NotFound);
```

Reply:
Ah, good catch. I've applied that to all the existing `map_err` calls